src/main/java/org/opentripplanner/street/model/edge/EscalatorEdge.java

```diff
@@ -0,0 +1,32 @@
+package org.opentripplanner.street.model.edge;
+
+import org.opentripplanner.framework.i18n.I18NString;
+import org.opentripplanner.street.model.vertex.Vertex;
+import org.opentripplanner.street.search.TraverseMode;
+import org.opentripplanner.street.search.state.State;
+
+/** Represents an escalator. An escalator edge can only be traversed by walking */
+public class EscalatorEdge extends Edge {
+
+  public EscalatorEdge(Vertex v1, Vertex v2) {
+    super(v1, v2);
+  }
+
+  @Override
+  public State[] traverse(State s0) {
+    // Only allow traversal by walking
+    if (s0.getNonTransitMode() == TraverseMode.WALK) {
```

Comment:
Is the getNonTransitMode WALK also when one is walking a bicycle or BICYCLE. Maybe it's just some flag set in the state data and the mode is still BICYCLE. I'm not sure. We should also block traversal on wheelchairs here.

Reply:
We think that the getNonTransitMode returns BICYCLE even when walking a bicycle so this code is probably right but the wheelchair should still be blocked somehow (there are probably examples in the street edge).